controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/JobStatus.java

```diff
@@ -98,6 +99,14 @@ else if (! lastTriggered.isPresent()) {
 
     /** Returns true unless this job last completed with a failure */
     public boolean isSuccess() { return ! jobError.isPresent(); }
+    
+    /** Returns true if last triggered is newer than last completed and was started after timeoutLimit */
+    public boolean isRunning(Instant timeoutLimit) {
```

Comment:
Isn't this the same as the existing `inProgress`, the only difference being that this supports a timeout limit? Consider merging the methods or at least using the same name for both.

Reply:
Good catch. Yes we should merge them and always pass a timeoutLimit.
I see that at least some of the uses of inProgress solved that problem separately so there is an opportunity for simplification here, however, I also notice there are no tests of those particular alternatives, so I need to add that first. So, this becomes a substantial change in itself, I want to do it in a separate PR so I added a TODO for now.

The other things are addressed.

Reply:
Great!